File: fillordontfill.c
#include <stdio.h>
#include <string.h>

int main(){

    int n;

    scanf ("%d", &n);

    char entrada[200];
    char ultimosNumeros[200];

    scanf ("%s", entrada);
    scanf ("%s", ultimosNumeros);

    int tamanho= strlen(ultimosNumeros);
    int tamanhoEntrada= strlen(entrada);

    int fit=1;
    int cont=0;
    for (int i=tamanhoEntrada- tamanho;i<tamanhoEntrada;i++){

        if (entrada[i] != ultimosNumeros[cont++]){

            fit=0;
            break;

        }
    }
    
    if (fit==1){
        printf ("Encaixa");
    } else{

        printf ("Não Encaixa");
    }
    return 0;
}

File: Fillordontfill.java
import java.util.Scanner;

public class Fillordontfill{


    public static void main (String[] args){

        Scanner scanner= new Scanner (System.in);

        String entrada= scanner.nextLine();

        String encaixa= scanner.nextLine();

        boolean fit= true;

        int tam= encaixa.length() -1;

        int contadorEncaixa=0;
        for (int i= entrada.length() -1 - tam; i<entrada.length(); i++){

            if (entrada.charAt(i)!=encaixa.charAt(contadorEncaixa++)){

               fit=false;
               break;
            } 
        }

        if (fit){

            System.out.println("Encaixa");
        } else {

            System.err.println("Não encaixa");
        }
    }
}